src/platform/graphics/render-target.js

```diff
@@ -135,7 +130,7 @@ class RenderTarget {
         }
 
         // render image flipped in Y
-        this.flipY = !!options.flipY;
+        this.flipY = options.flipY ?? false;
```

Comment:
not sure this one is better, I'd probably keep the old version which forces boolean.

Reply:
I was really going for consistency here since this was out of step with, say, the `depth` option. which doesn't coerce non-booleans into a boolean.

Reply:
For TypeScript developers, we can guarantee they're passing a boolean. So the code as is is totally correct in that case. However, for JS developers, you're basically advocating doing some level of work to sanitize function parameters. This isn't something the engine does much. I can find some instances. For example, check out this `Slot` code:

https://github.com/playcanvas/engine/blob/bd6256e83eadb065d4a5810959555200a9db4c9a/src/framework/components/sound/slot.js#L83-L89

Values are being explicitly converted to a `number` type where needed. What I would want is a set of explicit rules for the engine to follow to process passed in parameter values. And once we have those rules, we can apply them _everywhere_ and document them. Right now, the codebase isn't consistent.

Reply:
You can't really sanitize input, for example the "explicit" pitch code conversion you mention works for `NaN` but then already fails at `Infinity`, which happens rather often (division by zero somewhere deep down in the rabbit hole):

![image](https://user-images.githubusercontent.com/5236548/218693625-5173783c-484e-4162-bc4f-68f5daee4106.png)

Also this is "silent failing", so the code may work, but no one really knows why it happens to work for `NaN` but not for `Infinity` (until you debug for a certain amount of time). TypeScript doesn't really help either, it will happily pass `NaN`/`Infinity` around as numbers.

What actually works is generating greedy and pedantic type checks for literally every function that has JSDoc. But this is a performance concern while adding lots of extra code. So people making games in the PlayCanvas editor can always test everything with the type checking debug builds, but then e.g. only running the code in unchecking/fast release builds for the players.